src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -932,7 +932,57 @@ public void AndAlso()
 
             tokens.AddLast(QueryOperatorToken.And);
         }
+        
+        /// <summary>
+        ///   Wraps previous query with clauses and adds an AND operator to the given query
+        /// </summary>
+        public void AndAlso(bool wrapPreviousQueryClauses)
+        {
+            var tokens = GetCurrentWhereTokens();
+            
+            if (tokens.Last == null)
+                return;
 
+            if (tokens.Last.Value is QueryOperatorToken)
+                throw new InvalidOperationException("Cannot add AND, previous token was already an operator token.");
+            
+            if (wrapPreviousQueryClauses == false)
+            {
+                tokens.AddLast(QueryOperatorToken.And);
+            }
+            else
+            { 
+                tokens.AddFirst(OpenSubclauseToken.Create());
+                tokens.AddLast(CloseSubclauseToken.Create());
+                tokens.AddLast(QueryOperatorToken.And);
```

Comment:
Wouldn't it be better to add this code to `WrapQuery()` directly ? 
Or, just call this before we invoke `OnBeforeQuery` ?

Reply:
It needs to be in the DocumentQuery interface

Reply:
Why? We can handle that completely internally.
But I'm fine with adding that to the interface